Source/Core/Core/HW/WiiSave.cpp

```diff
@@ -29,35 +29,38 @@
 #include "Common/MsgHandler.h"
 #include "Common/NandPaths.h"
 #include "Common/StringUtil.h"
-#include "Common/Swap.h"
 
-const u8 CWiiSaveCrypted::s_sd_key[16] = {0xAB, 0x01, 0xB9, 0xD8, 0xE1, 0x62, 0x2B, 0x08,
-                                          0xAF, 0xBA, 0xD8, 0x4D, 0xBF, 0xC2, 0xA5, 0x5D};
-const u8 CWiiSaveCrypted::s_md5_blanker[16] = {0x0E, 0x65, 0x37, 0x81, 0x99, 0xBE, 0x45, 0x17,
-                                               0xAB, 0x06, 0xEC, 0x22, 0x45, 0x1A, 0x57, 0x93};
-const u32 CWiiSaveCrypted::s_ng_id = 0x0403AC68;
+constexpr u8 s_sd_key[16] = {0xAB, 0x01, 0xB9, 0xD8, 0xE1, 0x62, 0x2B, 0x08,
+                             0xAF, 0xBA, 0xD8, 0x4D, 0xBF, 0xC2, 0xA5, 0x5D};
+constexpr u8 s_md5_blanker[16] = {0x0E, 0x65, 0x37, 0x81, 0x99, 0xBE, 0x45, 0x17,
+                                  0xAB, 0x06, 0xEC, 0x22, 0x45, 0x1A, 0x57, 0x93};
+constexpr u32 s_ng_id = 0x0403AC68;
 
-bool CWiiSaveCrypted::ImportWiiSave(const std::string& filename)
+bool WiiSave::Import(const std::string& filename)
 {
-  CWiiSaveCrypted save_file(filename);
-  return save_file.m_valid;
+  WiiSave save_file{filename};
+  if (save_file.Import())
+  {
+    SuccessAlertT("Successfully imported save file(s)");
+    return true;
+  }
+  PanicAlertT("Import failed");
+  return false;
 }
 
-bool CWiiSaveCrypted::ExportWiiSave(u64 title_id)
+bool WiiSave::Export(u64 title_id)
 {
-  CWiiSaveCrypted export_save("", title_id);
-  if (export_save.m_valid)
+  WiiSave export_save{title_id};
+  if (export_save.Export())
   {
     SuccessAlertT("Successfully exported file to %s", export_save.m_encrypted_save_path.c_str());
```

Comment:
Thoughts on moving these dialogs to the frontend, so we can get more dialog/alert code out of Core, since the returned bool indicates success or failure?

It would involve also changing ExportAll to return a count of some sort as well. This could also be more user-friendly in the ExportAll case, since we could just show the total exported at the end. Currently for every save that fails to export in the ExportAll case, they'd get a dialog saying "Export failed" over and over, which could be annoying.

That said, this is a minor cleanup, so this just a suggestion.